src/pages/_middleware.ts

```diff
@@ -20,13 +23,37 @@ const PUBLIC_FILE = /\.(.*)$/
  * https://github.com/vercel/examples/edge-functions/ab-testing-simple
  *
  */
-export function middleware(req: NextRequest) {
-  // Only rewrite files that don't have a file extension (think favicon)
+export async function middleware(req: NextRequest) {
+  let response = NextResponse.next()
+
+  // think favicon etc
+  if (PUBLIC_FILE.test(req.nextUrl.pathname)) return response
+
   // Only rewrite if we are at the root
-  if (req.nextUrl.pathname === '/' && !PUBLIC_FILE.test(req.nextUrl.pathname)) {
+  if (req.nextUrl.pathname === '/') {
     // if there's a cookie with a token they are logged in
     let status = req.cookies[ACCESS_TOKEN_KEY] ? 'logged_in' : 'anon'
     req.nextUrl.pathname = status === 'logged_in' ? `/` : `/signup`
-    return NextResponse.rewrite(req.nextUrl)
+    response = NextResponse.rewrite(req.nextUrl)
+  } else if (req.nextUrl.pathname === '/signup') {
+    const cioId =
+      req.cookies[CIO_COOKIE_KEY] ||
+      req.nextUrl.searchParams.get(CIO_COOKIE_KEY)
+    if (cioId) {
+      const cioApiPath = `/customers/${cioId}/attributes`
+      const headers = new Headers({
+        Authorization: `Bearer ${process.env.CUSTOMER_IO_APPLICATION_API_KEY}`,
+      })
+      const request = new Request(`${CIO_BASE_URL}${cioApiPath}`, {headers})
+      const {customer} = await fetch(request).then((response) =>
+        response.json(),
+      )
+
+      if (customer?.attributes?.react_score > 1) {
```

Comment:
only addressed the React score here, but **any attribute in customer.io** can be used here for simple topic customization.

What's the algorithm?